src/javaharness/java/arcs/api/ParticleSpec.java

```diff
@@ -0,0 +1,51 @@
+package arcs.api;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ParticleSpec {
+    public final String name;
+    public final String implFile;
+    private final Map<String, HandleConnectionSpec> handleConnectionMap = new HashMap();
+
+    ParticleSpec(String name, String implFile) {
+        this.name = name;
+        this.implFile = implFile;
+    }
+
+    private void addHandleConnection(String name, String direction) {
+        this.handleConnectionMap.put(name, new HandleConnectionSpec(name, direction));
+    }
+
+    public boolean isInput(String name) {
+        HandleConnectionSpec connection = this.handleConnectionMap.get(name);
+        return connection.direction == "in" || connection.direction == "inout";
```

Comment:
use .equals(), not '=='. '==' in Java is reference equality only (equivalent to === JS)

Reply:
actually, here i switched to using the HandleConnectionSpec's methods instead of string comparison.